safe/impact_functions/core.py

```diff
@@ -203,3 +221,15 @@ def convert_to_old_keywords(converter, keywords):
                         break
             except KeyError:
                 pass
+
+
+def has_no_data(layer_data):
+    """Determine whether or not a layer contains nan values.
+
+    :param layer_data: Layer data that is to be inspected.
+    :type layer_data: ndarry
+
+    :return: The True if there is nodata in layer_data.
+    :rtype: bool
+    """
+    return numpy.isnan(numpy.sum(layer_data))
```

Comment:
This could be a nice addition to the IF base class. But as we will move to QGIS, we should think a fast similar way to do this in QGIS cc @timlinux @wonder-sk